.ci/no_old_refs.sh

```diff
@@ -5,27 +5,67 @@ set -e
 # Script requires GITHUB_TOKEN env variable
 MENTIONED_ISSUES=/tmp/mentioned_issues
 CLOSED_ISSUES=/tmp/failed_issues
+LINKED_ISSUES_FORMATTED=/tmp/linked_issues
+
+# Linked issues that are mentioned in the code
+LINKED_ISSUES_MENTIONED=/tmp/linked_issues_mentioned
 API_GITHUB_PREFIX="https://api.github.com/repos"
 GITHUB_HOST="https://github.com"
+CHECKSTYLE_ISSUE_PREFIX="https:\/\/github.com\/checkstyle\/checkstyle\/issues\/"
+MAIN_REPO="checkstyle/checkstyle"
+DEFAULT_BRANCH="master"
+
+# These are modified when event is of type pull_request
+if [ ! -z "$PR_HEAD_REPO_NAME" ]; then
+  MAIN_REPO=$PR_HEAD_REPO_NAME
+  DEFAULT_BRANCH=$GITHUB_HEAD_REF
+fi
 
 # collect issues where full link is used
-grep -IPohr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
-  | sed -e 's/.*github.com\///' >> $MENTIONED_ISSUES
+grep -IPonr "(after|[Tt]il[l]?) $GITHUB_HOST/[\w.-]+/[\w.-]+/issues/\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*github.com\//:/' >> $MENTIONED_ISSUES
 
 # collect checkstyle issues where only hash sign is used
-grep -IPohr "[Tt]il[l]? #\d{1,5}" . \
-  | sed -e 's/.*#/checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+grep -IPonr "[Tt]il[l]? #\d{1,5}" . \
+  | sed -e 's/:[a-zA-Z].*#/:checkstyle\/checkstyle\/issues\//' >> $MENTIONED_ISSUES
+
+# $LINKED_ISSUES need formatting before the are used
+if [ ! -z "$LINKED_ISSUES" ]; then
+  echo $LINKED_ISSUES | sed -e 's/,/\n/g' >> $LINKED_ISSUES_FORMATTED
+  sed -i "s/^/$CHECKSTYLE_ISSUE_PREFIX/" $LINKED_ISSUES_FORMATTED
+fi
+
+for MENTIONED_ISSUES_LINES in $(cat $MENTIONED_ISSUES); do
```

Comment:
Iteration variable can not be plural. Please rename

Reply:
MENTIONED_ISSUES should be rename to grep_lines

To mean that what content it has

Reply:
I feel `MENTIONED_ISSUES` is better than `grep_lines`, mentioned_issues describes the variables, grep is just a tool, changed it to `MENTIONED_ISSUES_LINE` i.e. singular form.

Reply:
Please print content of MENTIONED_ISSUES_LINE here

Reply:
@Vyom-Yadav, please do ∆∆

Reply:
Printed it.

Reply:
From logs:
./config/import-control.xml:107:checkstyle/checkstyle/issues/3511

Reply:
Yes it is of this nature -
```
./pom.xml:709:checkstyle/checkstyle/issues/2160
./pom.xml:1505:checkstyle/checkstyle/issues/5770
./pom.xml:1567:checkstyle/checkstyle/issues/11446
./.semaphore/semaphore.yml:56:checkstyle/checkstyle/issues/9807
./.semaphore/semaphore.yml:74:checkstyle/checkstyle/issues/9248
```
file_name:line_number:issue_url_suffix

Reply:
It is not mentioned issue, it is just output of grep tool.
So better to name variable to explain it's content.

Reply:
`grep_line` is vague, the line also passed through stream editor (sed), going by that logic we can even name it `sed_line`, if I looked it at in the future it would confuse me. The variable represents mentioned issue and where is it mentioned, if we want to include grep, we can maybe name it something like `MENTIONED_ISSUES_GREP_OUTPUT` or something of that form, what do you think?

Reply:
MENTIONED_ISSUES_GREP_OUTPUT this works for me, I just want to put in name message that content is not simple, it is more like structure

Reply:
MENTIONED_ISSUE_Structure
MENTIONing_package

Reply:
`MENTIONED_ISSUES_GREP_OUTPUT` sounds good, changed to that.